scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -837,6 +837,9 @@ def run(self):
                     self.defconfig['CONFIG_FAKE_ENTROPY_NATIVE_POSIX'] == 'y'):
                     instance.handler.seed = self.options.seed
 
+            if self.options.extra_test_args and instance.platform.name.startswith("native_posix"):
```

Comment:
I think platform unit_testing should also work

Reply:
I tried this on `tests/unit/base64` but the resulting `testbinary` doesn't seem to respond to any kind of CLI args. It does have `CONFIG_ZTEST_NEW_API=y` in its `prj.conf`, BTW

Reply:
@asemjonovs can you look into that? I thought that just adding `BOARD STREQUAL unit_testing` to the check so that `src/ztest_posix.c` is added to the build would have fixed that, but that doesn't seem to be correct.